File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if(nums.length < 1) {
            return 0; 
        }
        int maxSum = nums[0];
        int currentSum = maxSum; 
        for(int i = 1; i < nums.length; i++) {
            currentSum = Math.max(currentSum + nums[i],nums[i]); 
            maxSum =  Math.max(currentSum, maxSum); 
        }
        return maxSum; 
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
      ans = nums[0]
      curr_sum = 0
      
      for n in nums:
          if curr_sum < 0:
              curr_sum = 0
          curr_sum += n
          ans = max(ans, curr_sum)
      return ans
        
          
          
        